modin/core/dataframe/pandas/partitioning/partition_manager.py

```diff
@@ -848,28 +848,25 @@ def get_indices(cls, axis, partitions, index_func=None):
         -------
         pandas.Index
             A pandas Index object.
+        list of pandas.Index
+            The list of internal indices for each partition.
 
         Notes
         -----
         These are the global indices of the object. This is mostly useful
         when you have deleted rows/columns internally, but do not know
         which ones were deleted.
         """
+        if index_func is None:
+            index_func = lambda df: df.axes[axis]  # noqa: E731
         ErrorMessage.catch_bugs_and_request_email(not callable(index_func))
         func = cls.preprocess_func(index_func)
-        if axis == 0:
-            new_idx = (
-                [idx.apply(func) for idx in partitions.T[0]]
-                if len(partitions.T)
-                else []
-            )
-        else:
-            new_idx = (
-                [idx.apply(func) for idx in partitions[0]] if len(partitions) else []
-            )
+        target = partitions.T if axis == 0 else partitions
+        new_idx = [idx.apply(func) for idx in target[0]] if len(target) else []
         new_idx = cls.get_objects_from_partitions(new_idx)
         # TODO FIX INFORMATION LEAK!!!!1!!1!!
```

Comment:
I don't think there's any information leak here anymore :) I think the PR that added `get_objects_from_partitions` deleted it the access to `partitions[i]._data`. Could you delete this warning?

Reply:
I just double checked, and the info leak is still there in the `PandasOnRayDataframePartitionManager`:
https://github.com/modin-project/modin/blob/49c03981c2cc55495fa0f95e17e6fd2b6e421d9b/modin/core/execution/ray/implementations/pandas_on_ray/partitioning/partition_manager.py#L110

Reply:
I think it's doing this because `partition.get` will materialize partitions in serial, so it makes sense for it to do it, but the problem is if the partition has a call queue that will be ignored. Perhaps this is ok though?

Reply:
I dug into the history of this comment, and I still don't know what it's talking about. The diff from the original PR is [here](https://github.com/modin-project/modin/pull/721/files#diff-0c6caf26a29eab528096f123ec3ac58888509dcd47817c0a4d45a781e60ed06d). I don't see any information leak there, where the code uses `get()` to get the data.

Regarding the `_data` access in the ray partition manager, I think you're correct that it's incorrect because it doesn't drain the call queue. This use in `get_indices` happens to work because `apply` drain the call queue. I think we need to take a stance as to whether we should allow accessing the `_data` of  block/non-full-axis-virtual partitions. We should probably use `get()`, which drains the call queue, instead.

So back to this comment, I think the right thing would be to file an issue to drain the call queue before getting `_data` (or maybe just to get rid of all accesses to `_data`), and also to move this comment to `get_objects_from_partitions` until that issue is fixed. But I don't think that's a responsibility for this PR.

Reply:
Related issue #4530

Reply:
Sgtm - I'll go ahead and approve and merge then.

Reply:
Yeah, I think we should get rid of `._data` access anywhere but in the partition class itself (ideally).